website/src/models/group.js

```diff
@@ -168,6 +170,81 @@ schema.methods.isMember = function isGroupMember (user) {
   }
 };
 
+schema.methods.startQuest = async function startQuest (user) {
+  if (this.type !== 'party') throw new BadRequest('Must be a party to use this method');
+  if (!this.quest.key) throw new BadRequest('Party does not have a pending quest');
+  if (this.quest.active) throw new BadRequest('Quest is already active');
+
+  let userIsParticipating = this.quest.members[user._id];
+  let quest = questScrolls[this.quest.key];
+  let collected = {};
+  if (quest.collect) {
+    collected = _.transform(quest.collect, (result, n, itemToCollect) => {
+      result[itemToCollect] = 0;
+    });
+  }
+
+  let backgroundOperations = [];
+
+  this.markModified('quest');
+  this.quest.active = true;
+  if (quest.boss) {
+    this.quest.progress.hp = quest.boss.hp;
+    if (quest.boss.rage) this.quest.progress.rage = 0;
+  } else if (quest.collect) {
+    this.quest.progress.collect = collected;
+  }
+
+  // Changes quest.members to only include participating members
+  // TODO: is that important? What does it matter if the non-participating members
+  // are still on the object?
+  // TODO: is it important to run clean quest progress on non-members like we did in v2?
+  this.quest.members = _.pick(this.quest.members, _.identity);
+  let nonUserQuestMembers = _.without(_.keys(this.quest.members), user._id);
+
+  let members = await User.find(
+    { _id: { $in: nonUserQuestMembers } },
+    'party.quest items.quests auth.facebook auth.local preferences.emailNotifications',
+  ).exec();
+
+  if (userIsParticipating) {
+    members.unshift(user); // put participating user at the beginning of the array
+  }
+
+  _.each(members, (member) => {
+    member.party.quest.key = this.quest.key;
+    member.party.quest.progress.down = 0;
+    member.party.quest.collect = collected;
+    member.party.quest.completed = null;
+    member.markModified('party.quest');
+
+    if (this.quest.leader === member._id) {
+      member.items.quests[this.quest.key] -= 1;
+      member.markModified('items.quests');
+    }
+
+    if (member._id !== user._id) {
+      backgroundOperations.push(member.save());
```

Comment:
I'd say that as long as quests are only for parties just do one mass update that blocks the route